src/sdk/client.ts

```diff
@@ -41,17 +41,22 @@ export class NoslatedClient extends EventEmitter {
   useInspectorSet: Set<string>;
   platformEnvironmentVariables: root.noslated.IKeyValuePair[];
 
-  constructor() {
+  constructor(options?: NoslatedClientOptions) {
     super();
-    loggers.setSink(loggers.getPrettySink('sdk.log'));
     dumpConfig('sdk', config);
 
+    if (options?.logger) {
+      this.logger = options.logger;
+    } else {
+      loggers.setSink(loggers.getPrettySink('sdk.log'));
```

Comment:
The logger from the caller should be converted to a sink in order to cover all loggers created.

Reply:
`loggers.setSink` takes effect globally. In some special scenarios, such as DataPlane and SDK running in the same process, DataPlane sink settings will be overwritten. Currently, the sdk is not strongly dependent on the prefixed logger. I think that when the sdk is used as a lib format, it can accept external input to the logger instance.